### DISCUSSION TITLE
[Item] SCI论文阅读模版-new，自用生物医学类SCI笔记模板

### DISCUSSION THREAD

**Discussion by AlexShyXie** [windingwind/zotero-better-notes#501] [Note Templates]
我不喜欢其他人的那个表头和table，感觉太占位置了，所有参考别人的模板，以生物医学类的论文写作模块，重新修改了一点东西，效果如下：
![image](https://github.com/windingwind/zotero-better-notes/assets/49368033/c7fe0afd-aa4e-49f4-bc8a-d58199c855d1)
其中标题及摘要翻译是用pdf translator翻译之后获取的extra属性
使用的代码在下面，需要的自取
```
<!-- author:xiehui-->
<span style="color:#E91E63"><h1><font size="2">标题：${topItem.getField("title")}</font></h1></span>
<span style="color:#F8BBDO"><strong><u>作者：</u></strong>${topItem.getCreators().map((v)=>v.firstName+" "+v.lastName).join("; ")}</span><br/>
出版年份：${topItem.getField('date')}<br/><span style="background-color:#1976D2"></span>
<span style="background-color:#00ACC1">影响因子：${topItem.getField("libraryCatalog")}<br/></span>
<strong>标题及摘要翻译</strong>：${topItem.getField("extra")}<br/>
</span></h1>\n <h2 style="color:  #E65100; background-color:  #FFF8E1;">📜 前言</h2> <hr/>
<p></p>
</span></h1>\n <h2 style="color:#2E7D32; background-color:    #F1F8E9;">🔬 方法</h2> <hr/>
<p></p>
</span></h1>\n <h2 style="color:#4A148C; background-color:    #F5F5F5;">🚩 结果</h2> <hr/>
<p></p>
</span></h1>\n <h2 style="color: #006064; background-color:   #E0F7FA;">📌 Discussin & Conclusion & Limitation</h2> <hr/>
<p></p>
<h3>  <span style="color: #1565C0">创新：</span></h3>
<p></p>
<h3>  <span style="color: #1565C0">疑问：</span></h3>
<p></p>
</span></h1>\n <h2 style="color:#1565C0; background-color:   #E1F5FE;">🔬 我的评价</h2> <hr/>
```

**Comment by Tclassic1** [comment]
作者您好，我使用的zotero7使用这个代码的时候会报错：
Error: ReferenceError: topItem is not defined，您知道该如何解决吗

**Reply by windingwind** [reply]
请正确填写模板标题

**Reply by lisasuosuo** [reply]
> 作者您好，我使用的zotero7使用这个代码的时候会报错： Error: ReferenceError: topItem is not defined，您知道该如何解决吗

遇到了同样的问题，导致变量相关语句无法使用

**Reply by lisasuosuo** [reply]
> 请正确填写模板标题

搞定了，模版标题要用[Item]开头

**Comment by qwpyyx** [comment]
很有用，感谢

**Comment by windingwind** [comment]
The original post is the template content that can be pasted into the template editor of BN. The latest version supports `import template from clipboard`, here is the full template for importing:

```yaml
# This template is specifically for importing/sharing, using better 
# notes 'import from clipboard': copy the content and
# goto Zotero menu bar, click Edit->New Template from Clipboard.  
# Do not copy-paste this to better notes template editor directly.
name: "[Item] SCI论文阅读模版-new，自用生物医学类SCI笔记模板"
content: |-
  // @author xiehui
  // @link https://github.com/windingwind/zotero-better-notes/discussions/501
  <span style="color:#E91E63"><h1><font size="2">标题：${topItem.getField("title")}</font></h1></span>
  <span style="color:#F8BBDO"><strong><u>作者：</u></strong>${topItem.getCreators().map((v)=>v.firstName+" "+v.lastName).join("; ")}</span><br/>
  出版年份：${topItem.getField('date')}<br/><span style="background-color:#1976D2"></span>
  <span style="background-color:#00ACC1">影响因子：${topItem.getField("libraryCatalog")}<br/></span>
  <strong>标题及摘要翻译</strong>：${topItem.getField("extra")}<br/>
  </span></h1>\n <h2 style="color:  #E65100; background-color:  #FFF8E1;">📜 前言</h2> <hr/>
  <p></p>
  </span></h1>\n <h2 style="color:#2E7D32; background-color:    #F1F8E9;">🔬 方法</h2> <hr/>
  <p></p>
  </span></h1>\n <h2 style="color:#4A148C; background-color:    #F5F5F5;">🚩 结果</h2> <hr/>
  <p></p>
  </span></h1>\n <h2 style="color: #006064; background-color:   #E0F7FA;">📌 Discussin & Conclusion & Limitation</h2> <hr/>
  <p></p>
  <h3>  <span style="color: #1565C0">创新：</span></h3>
  <p></p>
  <h3>  <span style="color: #1565C0">疑问：</span></h3>
  <p></p>
  </span></h1>\n <h2 style="color:#1565C0; background-color:   #E1F5FE;">🔬 我的评价</h2> <hr/>
```

**Reply by brightsongs** [reply]
太棒了，不过Discussin有一个拼写错误

**Comment by cutebird00** [comment]
你好！非常感谢，我好喜欢这个模板！我有一个小问题是，我该怎么把它作为这个文献的独立笔记建出来呢？或者说是插入这里。
![图片](https://github.com/windingwind/zotero-better-notes/assets/138377212/e2da9300-36ac-42bb-a78e-8bafa88bd2dc)
我好像只能把它插入到工作区笔记中。
![图片](https://github.com/windingwind/zotero-better-notes/assets/138377212/2f076efc-7537-4ff9-a4c7-9fcbc410be62)

**Reply by AlexShyXie** [reply]
这里
![image](https://github.com/windingwind/zotero-better-notes/assets/49368033/9e8803aa-642c-4505-8d2b-d6c708deefea)

**Comment by Uoyiz00** [comment]
您好，我复制这个代码保存后，点击预览以及新建时会报错：Error: ReferenceError: topItem is not defined，您知道该如何解决吗
!![image](https://github.com/windingwind/zotero-better-notes/assets/142001111/97787ac9-eefc-413b-98eb-0e746d7e8296)
以及楼上说改模板标题我没get到，是改哪里呢

**Reply by AlexShyXie** [reply]
这是better notes的模板，不是zotcard的

**Comment by Yibing33** [comment]
请问出现这种情况要怎么解决呢？
![微信截图_20230815154841](https://github.com/windingwind/zotero-better-notes/assets/142248482/96193306-3d47-473d-aa74-8c2abb8189ea)
![微信截图_20230815154855](https://github.com/windingwind/zotero-better-notes/assets/142248482/6c929c4b-9638-4607-8a39-3b6700a1a2aa)

**Reply by AlexShyXie** [reply]
你的条目没有title？

**Reply by Yibing33** [reply]
条目没有title是什么意思啊？是文章标题吗？在软件内标题什么的都是正常显示的

**Reply by XHHHX** [reply]
请问问题解决了吗？遇到了同样的问题

**Reply by sartre123** [reply]
模版名称要加“[item]”

**Comment by AbysSses** [comment]
![image](https://github.com/windingwind/zotero-better-notes/assets/136069464/74213a2c-3846-442f-8673-3f8974fb749c)
我的插入后不抓取作者年份影响因子以及翻译什么的是什么原因啊

**Reply by AlexShyXie** [reply]
你这个是条目吗？对着文件进行的操作

**Reply by Helen-13-14** [reply]
我的也是这样

**Reply by janeluyip** [reply]
我也这样

**Comment by yuanyuanfka** [comment]
怎么下载啊

**Comment by Kieran0526** [comment]
请问这个摘要翻译的部分如何显示呢

**Reply by bionoob7** [reply]
一样的问题，请问您解决了

**Comment by tianlexinyou** [comment]
请问楼上说的“搞定了，模版标题要用[Item]开头”是什么意思啊，代码具体要怎么修改呢，为啥我把所有的topItem替换成Item，还是显示Error: ReferenceError: Item is not defined呀，谢谢楼主

**Reply by leanondaboat** [reply]
不是把代码里的topItem改成Item，是把模板名称改成[Item]开头就可以了

**Reply by tianlexinyou** [reply]
谢谢！终于好用了！太感谢了！！

**Reply by zhouqian177** [reply]
![image](https://github.com/windingwind/zotero-better-notes/assets/7477793/77a2350a-0cc5-41c3-86a2-c77e8bce72f7)
这样就可以了

**Comment by guojianrongtj** [comment]
能改进为仅显示前三作者或者第一作者+通讯作者就好了

**Reply by Kingxs840** [reply]
有了t我一下

**Comment by AmethystAmorFati** [comment]
![屏幕截图 2024-01-25 154253](https://github.com/windingwind/zotero-better-notes/assets/155733034/a2041969-4ec0-4c28-a483-c437b6b6616b)
请问这样是为什么

**Comment by Gearless-Joe** [comment]
请问能不能设置成标题提取条目标题而非PDF标题？
因为感觉一般操作习惯是看着文献打开笔记，这时候笔记中的“标题”行会自动提取PDF的标题，而PDF的标题常常奇奇怪怪的。条目已经提取好的标题往往是正常的，所以如果能让笔记模板直接提取条目标题就好了，感谢！
